standalone/runtime/tests/test_issue.rs

```diff
@@ -494,182 +572,411 @@ mod execute_refund_payment_limits {
+                    ..CoreVaultData::vault(VAULT)
+                },
+            );
 
-        execute_issue(issue_id);
+            let (issue_id, issue) = request_issue(requested_btc);
+            let sent_btc = (issue.amount + issue.fee) * 4;
+            let post_request_state = ParachainState::get();
 
-        assert_eq!(
-            ParachainState::get(),
-            post_liquidation_status.with_changes(|user, _vault, liquidation_vault, _fee_pool| {
-                user.free_tokens += issue.amount;
+            ExecuteIssueBuilder::new(issue_id)
+                .with_amount(sent_btc)
+                .assert_execute();
+
+            // not enough collateral to back sent amount, so it's as if the user sent the correct amount
+            assert_eq!(
+                ParachainState::get(),
+                post_request_state.with_changes(|user, vault, _, fee_pool| {
+                    user.locked_balance -= issue.griefing_collateral;
+                    user.free_balance += issue.griefing_collateral;
+
+                    user.free_tokens += issue.amount;
+                    fee_pool.vault_rewards += issue.fee;
+
+                    vault.issued += issue.fee + issue.amount;
+                    vault.to_be_issued -= issue.fee + issue.amount;
+                })
+            );
 
-                // since the vault has been liquidated execute rewards
-                // are transferred to the treasury
+            let refund_id = assert_refund_request_event();
+            let refund = RefundPallet::get_open_refund_request_from_id(&refund_id).unwrap();
+            assert_eq!(refund.issue_id, issue_id);
 
-                user.free_balance += issue.griefing_collateral;
-                user.locked_balance -= issue.griefing_collateral;
+            let user_issues = IssuePallet::get_issue_requests_for_account(account_of(USER));
+            let (_, onchain_issue) = user_issues.iter().find(|(id, _)| id == &issue_id).unwrap();
+            assert_eq!(onchain_issue.status, IssueRequestStatus::Completed(Some(refund_id)));
+        });
+    }
 
-                liquidation_vault.to_be_issued -= issue.amount + issue.fee;
-                liquidation_vault.issued += issue.amount + issue.fee;
-            })
-        );
-    });
+    #[test]
+    fn integration_test_issue_execute_postcond_liquidated() {
+        test_with_initialized_vault(|| {
+            let (issue_id, issue) = RequestIssueBuilder::new(10_000).request();
+
+            drop_exchange_rate_and_liquidate(VAULT);
+            let post_liquidation_status = ParachainState::get();
+
+            execute_issue(issue_id);
+
+            assert_eq!(
+                ParachainState::get(),
+                post_liquidation_status.with_changes(|user, _vault, liquidation_vault, fee_pool| {
+                    user.free_tokens += issue.amount;
+                    // fee_pool.vault_rewards += issue.fee; // this is not the case - verify that's
```

Comment:
I'm not sure about this - the code seems to add vault rewards unconditionally: https://github.com/interlay/interbtc/blob/master/crates/issue/src/lib.rs#L445
However, the test fails if this assertion is enabled.

Reply:
Ah yes, the problem here is that if there is only one Vault and if it is liquidated all rewards are sent to the treasury.